eng/Directory.Build.Common.props

```diff
@@ -29,7 +29,7 @@
 
   <!-- Setup default project properties -->
   <PropertyGroup>
-    <LangVersion>11.0</LangVersion>
+    <LangVersion>12.0</LangVersion>
```

Comment:
Should we just bump this to `latest`?   That would be my vote.   We can always use a specific version if we have a reason to pin at some point, but generally we want to take advantage of whatever the repository's associated SDK supports.

Reply:
Yeah I can bump this to latest. Although I was getting a failure when I added the target but didn't upgrade the language version so I think there are some safeguards in place to keep it in sync

Reply:
The string "latest" would be equivalent to bumping to 12.0.   It's basically the shortcut for keeping it up to date.